Makefile

```diff
@@ -16,6 +16,7 @@ init: $(DLEQ_LIB)
 lint: init
 	./scripts/install-lint.sh
 	${GOPATH}/bin/golangci-lint run
+	shellcheck scripts/*.sh
```

Comment:
We need linting/formatting for the various source types in the `ui` folder too, but I'm not taking that on in this PR.

Reply:
do we need shellcheck here? I'd rather put it under `lint` so that we don't force users to install it

Reply:
It is under the `lint` target right now, which only developers would be invoking. After reading your comment, I made the `lint` and `format` targets more granular.  If you are only changing Go code, you can now invoke `make lint-go` to only check the go code without installing `shellcheck`.  CI will hopefully run the go checks, shell checks and future UI code checks.

Reply:
sorry, i think i misread it previously. this looks good to me!